sub-groups/index.md

```diff
@@ -0,0 +1,36 @@
+# Basic sub-group extension
```

Comment:
General comment: Seems like this proposal is missing device info properties?

`info::device::max_num_sub_groups` and `info::device::sub_group_independent_forward_progress`